### PR TITLE
feat: adds waf tutorial view

### PR DESCRIPTION
**Pull request by kendallstrautman**
## 🔗 Relevant links

- [Preview link](https://dev-portal-git-ksadds-waf-tutorial-view-hashicorp.vercel.app/well-architected-framework/com/cloud-operating-model) 🔎
- [Asana task](https://app.asana.com/0/1199634971449915/1202557239529686) 🎟️
- [Figma File](https://www.figma.com/file/VD7ahvXuXWJApeGnhbW4hv/Developer?node-id=11472%3A114805) 🎨 

## 🗒️ What

This PR adds the WAF tutorial view. 

## 🛠️ How

Reusing lots of components and logic from the current product tutorial view. 

## 📸 Design Screenshots
Screen Shot 2022-08-04 at 4 56 55 PM

## 🧪 Testing

- [Visit a tutorial page](https://dev-portal-git-ksadds-waf-tutorial-view-hashicorp.vercel.app/well-architected-framework/com/cloud-operating-model)
- ensure it renders like the other tutorial pages

## 💭 Anything else?

I'm still going to have a round of refinements in a follow up PR.

### CODE REVIEW

File: src/pages/well-architected-framework/[...tutorialSlug]/index.tsx [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+import { buildCategorizedWafSidebar } from 'views/well-architected-framework/utils/generate-sidebar-items'
+import WellArchitectedFrameworkTutorialView from 'views/well-architected-framework/tutorial-view'
+import {
+	Collection as ApiCollection,
+	TutorialLite as ApiTutorialLite,
+} from 'lib/learn-client/types'
+import { generateTopLevelSidebarNavData } from 'components/sidebar/helpers'
+import { MenuItem, SidebarProps } from 'components/sidebar'
+import { generateWafCollectionSidebar } from 'views/well-architected-framework/utils/generate-collection-sidebar'
+import { getNextPrevious } from 'views/tutorial-view/components'
```
Comment (kendallstrautman) [inline src/pages/well-architected-framework/[...tutorialSlug]/index.tsx:18]: reorder these

File: src/pages/well-architected-framework/[...tutorialSlug]/index.tsx [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+
+export async function getStaticProps({ params }) {
```
Comment (kendallstrautman) [inline src/pages/well-architected-framework/[...tutorialSlug]/index.tsx:20]: type this

File: src/pages/well-architected-framework/[...tutorialSlug]/index.tsx [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+		{
+			title: fullTutorialData.name,
+			url: `/${collectionContext.current.slug}/${splitProductFromFilename(
+				fullTutorialData.slug
+			)}`,
+			isCurrentPage: true,
+		},
+	]
+
+	const lastTutorialIndex = collectionContext.current.tutorials.length - 1
```
Comment (kendallstrautman) [inline src/pages/well-architected-framework/[...tutorialSlug]/index.tsx:88]: Separate this into own function?

File: src/views/tutorial-view/components/next-previous/helpers.ts [hunk 4]
```diff
@@ -3,22 +3,23 @@ import {
...
-	getCollectionSlug,
-	getTutorialSlug,
-} from 'views/collection-view/helpers'
 import { NextPreviousProps } from '.'
 
 interface GetNextPreviousParams {
 	currentTutorialSlug: string
 	currentCollection: ClientCollection
 	nextCollectionInSidebar: ClientCollectionLite
+	formatting: {
```
Comment (kendallstrautman) [inline src/views/tutorial-view/components/next-previous/helpers.ts:12]: The waf tutorials and collections have a different slug structure, introducing this to handle that.

File: src/views/well-architected-framework/collection-view/index.tsx [hunk 5]
```diff
@@ -25,16 +26,10 @@ export default function WellArchitectedFrameworkCollectionView({
...
-					title: metadata.wafName,
-					levelButtonProps: {
-						levelUpButtonText: `Main Menu`,
-						levelDownButtonText: 'Previous',
-					},
-					overviewItemHref: `/${metadata.wafSlug}`,
-					menuItems: sidebarSections,
-					showFilterInput: false,
-				},
+				generateWafCollectionSidebar(
```
Comment (kendallstrautman) [inline src/views/well-architected-framework/collection-view/index.tsx:29]: Made a reusable function as this level is repeated through multiple views.

File: src/views/well-architected-framework/tutorial-view/index.tsx [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+ * - check design spec for any inconsistencies
+ * - vet the links
+ * - check on the tutorial rewrite plugin, to accept waf links
+ */
+
+export default function WellArchitectedFrameworkTutorialView({
+	tutorial,
+	metadata,
+	layoutProps,
+}) {
```
Comment (kendallstrautman) [inline src/views/well-architected-framework/tutorial-view/index.tsx:26]: type this

File: src/pages/well-architected-framework/[...tutorialSlug]/index.tsx [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+		t.slug.endsWith(tutorialFilename)
+	)
+
+	// The tutorial doesn't exist in collection - return 404
+	if (!currentCollection || !currentTutorialReference) {
+		return null
+	}
+
+	// Need to get tutorial data with full mdx content
+	const fullTutorialData = await getTutorial(currentTutorialReference.slug)
```
Comment (kendallstrautman) [inline src/pages/well-architected-framework/[...tutorialSlug]/index.tsx:39]: There's a _little_ bit of duplication between here and the product tutorial prop gen. But its different enough to make me think this is fine 👌

**Review by zchsh — APPROVED** [review]
🙌 This looks great to me!

+1 that a little duplication between the "standard" and "waf" tutorials views seems fine - i definitely prefer that to forcing an abstraction 👍

**Review by ashleemboyer — APPROVED** [review]
🥇

**Pull request merged**